### PR TITLE
Remove usage of setNativeProps

### PR DESCRIPTION
**Pull request by rushatgabhane**
### Why?

`setNativeProps` will not be supported in the post-Fabric world ([docs](https://reactnative.dev/docs/new-architecture-library-intro#migrating-off-setnativeprops)). Removing it's usage is a prerequisite to enable Fabric in https://github.com/Expensify/App/issues/8503 

### Details

For SignInPageForm, TextInputLabel/index and TextInput/index
Usage of `setNativeProps` is migrated to `setAttribute`

For ReportActionCompose, Form, OptionsSelector and ReportSettingsPage
Usage of `setNativeProps` is migrated to state

For InvertedFlatList
Usage of `setNativeProps` is migrated by passing styles`.translate0InvertY` as a prop

For Picker and ReportActionItemMessageEdit
Usage of `setNativeProps` is just removed. No migration was needed.

RoomNameInput was modified to support controlled inputs using `value` prop.

### Fixed Issues

$ https://github.com/Expensify/App/issues/8503
(prerequisite step)
###  Tests 
Lotsss of tests here. Brace yourself!

### Sign In page 

1. Navigate to NewDot on the web.
2. Right-click on the email field.
3. Click on the 1Password option to show the autofill.
4. 1Password Autofill option should be shown on the input.

### New group page / OptionsSelector
1. Click on green fab icon (+)
2. New group 
3. Type the email address of a contact that you have.
4. Select the contact
5. Verify that the entered **text** is selected.

image

### Message Edit

1. Navigate to a chat in New expensify
7. Send a message.
8. Hover over the message and click on the edit comment option in the menu
9. Verify that the cursor is set to the end of the message.

And
On web and desktop
1. Navigate to a chat in New expensify
2. Send a message.
3. Tap up arrow key to edit the previous comment.
4. See the cursor is set to the end of the message.

### Picker
1. Run `npm run storybook` and test picker functionalities. Make sure that we can edit the input and all values are updated accordinly.
2. Test ALL usages of Picker (any dropdown select input) in the app. Set new value, update value and verify error states

- Settings > Payments > Add payment method > Add debit card > Select state
- Workspace > Connect bank account (Additional information) > State
- Policy room (#announce) > Settings > Notify me about new messages
- Settings > Preferences > Priority mode
- Settings > Preferences > Language
- Settings > Profile > Preferred pronouns
- Settings > Profile > Timezone
- Workspace > Connect bank account (Company step) > Address state
- Workspace > Connect bank account (Company step) > Company type
- Workspace > Connect bank account (Company step) > Corporation state
- Workspace > Connect bank account (Personal information) > State
- Login page > Select locale (below terms and conditions)

### Form component
1. Run `npm run storybook` and test Form. Make sure that we can edit the input and all values are updated accordingly.
2. Verify that form inputs validate on Blur and Submit.
10. Verify that the correct values are being submitted.

### Address search
1. Go to Settings - Payments -> Add payment method -> Add debit card page
2. Verify that you can search an address.
3. Verify that you can select an address.

### Pages using Form.js
1. Add debit card page
    1. Go to Settings - Payments -> Add payment method -> Add debit card page
    2. Verify that you can edit all inputs.
    3. Verify that the input is being validated on blur, and submit.
2. Request a call page
    1. Go to Concierge -> Click on the call button 
    2. Verify that you can edit all inputs.
    3. Verify that the input is being validated on blur, and submit.
3. Profile page
   1. Go to settings -> Profile 
   2. Verify that you can edit all inputs.
   3. Verify that the input is being validated on blur, and submit.

### Chat / InvertedFlatList
On Safari and Firefox,
1. Verify that the scrolling a chat isn't "choppy"

### Composer

1. Go to any chat
2. Verify that you can type text in the composer.
3. Open emoji picker
4. Verify that you can add emojis, and that cursor postion is after emoji.

### This step is **_not_** applicable for QA - Room name input
Prerequisite - comment out the validations in [ReportSettingsPage](https://github.com/Expensify/App/blob/3cb5dbf159bf29b7d4a563ea75450844d8567ad6/src/pages/ReportSettingsPage.js#L146-L161) 

1. Click on green fab icon (+)
7. Create a new room with the name "foo"
8. Repeat step 1, and create a new room with the name "bar" in the same workspace as room "foo".
9. Go to room "bar" -> Room settings
10. Rename it to "foo"
11. Click Save
12. Click the "X" mark after RBR error.
13. Verify that room name input is now back to "bar".

https://user-images.githubusercontent.com/29673073/189484614-1ace179e-4424-4221-b708-54bab61d509a.mov

- [ ] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [ ] I added steps for Staging and/or Production testing in the `QA steps` section
    - [ ] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I ran the tests on **all platforms** & verified they passed on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [x] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The Contributor+ will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### Sign In page 

1. Navigate to NewDot on the web.
2. Right-click on the email field.
3. Click on the 1Password option to show the autofill.
4. 1Password Autofill option should be shown on the input.

### New group page / OptionsSelector
1. Click on green fab icon (+)
2. New group 
3. Type the email address of a contact that you have.
4. Select the contact
5. Verify that the entered **text** is selected.

image

### Message Edit

1. Navigate to a chat in New expensify
7. Send a message.
8. Hover over the message and click on the edit comment option in the menu
9. Verify that the cursor is set to the end of the message.

And
On web and desktop
1. Navigate to a chat in New expensify
2. Send a message.
3. Tap up arrow key to edit the previous comment.
4. See the cursor is set to the end of the message.

### Picker
1. Got to storybook and test picker functionalities. Make sure that we can edit the input and all values are updated accordinly.
2. Test ALL usages of Picker (any dropdown select input) in the app. Set new value, update value and verify error states

- Settings > Payments > Add payment method > Add debit card > Select state
- Workspace > Connect bank account (Additional information) > State
- Policy room (#announce) > Settings > Notify me about new messages
- Settings > Preferences > Priority mode
- Settings > Preferences > Language
- Settings > Profile > Preferred pronouns
- Settings > Profile > Timezone
- Workspace > Connect bank account (Company step) > Address state
- Workspace > Connect bank account (Company step) > Company type
- Workspace > Connect bank account (Company step) > Corporation state
- Workspace > Connect bank account (Personal information) > State
- Login page > Select locale (below terms and conditions)

### Form component
1. Go to storybook and test Form. Make sure that we can edit the input and all values are updated accordingly.
2. Verify that form inputs validate on Blur and Submit.
10. Verify that the correct values are being submitted.

### Pages using Form.js
1. Add debit card page
    1. Go to Settings - Payments -> Add payment method -> Add debit card page
    2. Verify that you can edit all inputs.
    3. Verify that the input is being validated on blur, and submit.
4. Request a call page
    1. Go to Concierge -> Click on the call button 
    2. Verify that you can edit all inputs.
    3. Verify that the input is being validated on blur, and submit.
11. Profile page
   1. Go to settings -> Profile 
   2. Verify that you can edit all inputs.
   3. Verify that the input is being validated on blur, and submit.

### Chat / InvertedFlatList
On Safari and Firefox,
1. Verify that the scrolling a chat isn't "choppy"

### Composer

1. Go to any chat
2. Verify that you can type text in the composer.
3. Open emoji picker
4. Verify that you can add emojis, and that cursor postion is after emoji.

- [x] Verify that no errors appear in the JS console

### CODE REVIEW

**Review by rushatgabhane — COMMENTED** [review]
Update: This is fixed - https://github.com/Expensify/App/pull/10934#issuecomment-1242850659

<hr />

Bug - `AddressSearch` when used with `Form` is broken. (https://expensify.slack.com/archives/C01GTK53T8Q/p1663164808524219)

1. Go to Settings -> Payments -> Add payment method -> Add debit card
2. Search for any billing address
3. Click a suggested address

Expected: Street address, City, State and Zipcode are autofilled.
Actual: Nothing happens

**Review by parasharrajat — CHANGES_REQUESTED** [review]
Just a few thoughts. I will start my review soon. There a lot of things to go over.

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Comment by luacmartins** [comment]
@parasharrajat please let me know once you are done reviewing this one.

**Comment by luacmartins** [comment]
@parasharrajat hmm I see that this PR is only a part of the linked issue. Do you intend to work on the full migration to Fabric?

File: src/pages/home/report/ReportActionCompose.js [hunk 1]
```diff
@@ -372,9 +369,9 @@ class ReportActionCompose extends React.Component {
      * @param {String} newComment
      */
     updateComment(newComment) {
-        this.textInput.setNativeProps({text: newComment});
         this.setState({
             isCommentEmpty: !!newComment.match(/^(\s|`)*$/),
+            value: newComment,
```
Comment (luacmartins) [inline src/pages/home/report/ReportActionCompose.js:374]: We should add `value` to [state](https://github.com/Expensify/App/blob/59019d53723163e99b5ddeb1968dd53a2f1b433f/src/pages/home/report/ReportActionCompose.js#L136-L147)

File: src/pages/home/report/ReportActionItemMessageEdit.js [hunk 2]
```diff
@@ -97,7 +97,6 @@ class ReportActionItemMessageEdit extends React.Component {
      * @param {String} newDraft
      */
     updateDraft(newDraft) {
-        this.textInput.setNativeProps({text: newDraft});
```
Comment (luacmartins) [inline src/pages/home/report/ReportActionItemMessageEdit.js:100]: Let's make sure [this issue](https://github.com/Expensify/App/pull/3510) doesn't come back

File: src/pages/settings/Profile/ProfilePage.js [hunk 3]
```diff
@@ -291,7 +291,6 @@ class ProfilePage extends Component {
                             items={timezones}
                             isDisabled={this.state.isAutomaticTimezone}
                             defaultValue={this.state.selectedTimezone}
-                            value={this.state.selectedTimezone}
```
Comment (luacmartins) [inline src/pages/settings/Profile/ProfilePage.js:294]: Can we remove other usages of `this.state.selectedTimezone` in this file?

File: src/styles/styles.js [hunk 4]
```diff
@@ -2671,6 +2671,10 @@ const styles = {
         fontFamily: fontFamily.GTA,
         marginLeft: 6,
     },
+
+    translate0InvertY: {
+        transform: 'translate3d(0,0,0) scaleY(-1)',
```
Comment (luacmartins) [inline src/styles/styles.js:2676]: I think this is the correct format?

```suggestion
        transform: [{translateX: 0}, {translateY: 0}, {scaleY: -1}],
```

**Review by luacmartins — COMMENTED** [review]
I think it would have been better to break this changes into multiple PRs so it's easier to test them. I'll test once the change requests have been addressed.

**Pull request closed without merging**